Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj

```diff
@@ -174,6 +177,22 @@
 		F4FED324235823A3003C139C /* NSPasteboardAdditions.mm in Sources */ = {isa = PBXBuildFile; fileRef = F4FED3222358215E003C139C /* NSPasteboardAdditions.mm */; };
 /* End PBXBuildFile section */
 
+/* Begin PBXBuildRule section */
+		E372BC2C2C08C74A006DFE67 /* PBXBuildRule */ = {
+			isa = PBXBuildRule;
+			compilerSpec = com.apple.compilers.proxy.script;
+			filePatterns = "*.appex";
+			fileType = pattern.proxy;
+			inputFiles = (
+			);
+			isEditable = 1;
+			outputFiles = (
+				"$(BUILT_PRODUCTS_DIR)/WebKitTestRunnerApp.app/Extensions/$(INPUT_FILE_NAME)",
+			);
+			script = "if [[ ${WK_PLATFORM_NAME} == \"macosx\" ]]; then\n    exit\nfi\n\nDESTINATION_PATH=${BUILT_PRODUCTS_DIR}/WebKitTestRunnerApp.app/Extensions/${INPUT_FILE_NAME}\nditto ${INPUT_FILE_PATH} ${DESTINATION_PATH}\n\n/usr/libexec/PlistBuddy -c \"Set :CFBundleIdentifier ${PRODUCT_BUNDLE_IDENTIFIER}.${INPUT_FILE_BASE}\" \"${DESTINATION_PATH}/Info.plist\"\n/usr/libexec/PlistBuddy -c \"Delete :EXAppExtensionAttributes:EXExtensionPointIdentifier dict\" \"${DESTINATION_PATH}/Info.plist\"\n\nif [[ ${INPUT_FILE_BASE} == \"WebContentExtension\" ]]; then\n    EXTENSION_POINT_ID=\"com.apple.web-browser-engine.content\"\nelif [[ ${INPUT_FILE_BASE} == \"WebContentCaptivePortalExtension\" ]]; then\n    EXTENSION_POINT_ID=\"com.apple.web-browser-engine.content\"\nelif [[ ${INPUT_FILE_BASE} == \"GPUExtension\" ]]; then\n    EXTENSION_POINT_ID=\"com.apple.web-browser-engine.rendering\"\nelif [[ ${INPUT_FILE_BASE} == \"NetworkingExtension\" ]]; then\n    EXTENSION_POINT_ID=\"com.apple.web-browser-engine.networking\"\nfi\n\n/usr/libexec/PlistBuddy -c \"Add :EXAppExtensionAttributes:EXExtensionPointIdentifier string ${EXTENSION_POINT_ID}\" \"${DESTINATION_PATH}/Info.plist\"\n";
```

Comment:
Two small suggestions:

- Add `set -e` to the very beginning, so that failures from `ditto` or PlistBuddy stop the build.
- Remove the `if [[ ${WK_PLATFORM_NAME} == "macosx" ]]; then exit` clause. Exiting without creating the output means we're lying to the build system, because it expects the rule to produce this output unconditionally. Instead, you can add `*.appex` to EXCLUDED_SOURCE_FILE_NAMES here https://github.com/WebKit/WebKit/blob/4d53fb735617548d3f4402459fb426c20d232c1d/Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig#L40, and that will prevent it from scheduling the rule at all.

Reply:
Ah, good points! I will address these issues in the upcoming patch.

Thanks for reviewing!